sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveWindowFunctionQuerySuite.scala

```diff
@@ -0,0 +1,847 @@
+
+  createQueryTest("windowing.q -- 39. testUDFOnOrderCols",
+    """
+      |select p_mfgr, p_type, substr(p_type, 2) as short_ptype,
+      |rank() over (partition by p_mfgr order by substr(p_type, 2))  as r
+      |from part
+    """.stripMargin, reset = false)
+
+  createQueryTest("windowing.q -- 40. testNoBetweenForRows",
+    """
+      |select p_mfgr, p_name, p_size,
+      |sum(p_retailprice) over (distribute by p_mfgr sort by p_name rows unbounded preceding) as s1
+      |from part
+    """.stripMargin, reset = false)
+
+  createQueryTest("windowing.q -- 41. testNoBetweenForRange",
+    """
+      |select p_mfgr, p_name, p_size,
+      |sum(p_retailprice) over (distribute by p_mfgr sort by p_size range unbounded preceding) as s1
+      |from part
+    """.stripMargin, reset = false)
+
+  createQueryTest("windowing.q -- 42. testUnboundedFollowingForRows",
+    """
+      |select p_mfgr, p_name, p_size,
+      |sum(p_retailprice) over (distribute by p_mfgr sort by p_name
+      |rows between current row and unbounded following) as s1
+      |from part
+    """.stripMargin, reset = false)
+
+  createQueryTest("windowing.q -- 43. testUnboundedFollowingForRange",
+    """
+      |select p_mfgr, p_name, p_size,
+      |sum(p_retailprice) over (distribute by p_mfgr sort by p_size
+      |range between current row and unbounded following) as s1
+      |from part
+    """.stripMargin, reset = false)
+
+  createQueryTest("windowing.q -- 44. testOverNoPartitionSingleAggregate",
+    """
+      |select p_name, p_retailprice,
+      |round(avg(p_retailprice) over(),2)
+      |from part
+      |order by p_name
+    """.stripMargin, reset = false)
+}
+
+class HiveWindowFunctionQueryWithoutCodeGenSuite extends HiveWindowFunctionQueryBaseSuite {
+  var originalCodegenEnabled: Boolean = _
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    originalCodegenEnabled = conf.codegenEnabled
+    sql("set spark.sql.codegen=false")
+  }
+
+  override def afterAll(): Unit = {
+    sql(s"set spark.sql.codegen=$originalCodegenEnabled")
+    super.afterAll()
+  }
+}
+
+abstract class HiveWindowFunctionQueryFileBaseSuite
+  extends HiveCompatibilitySuite with BeforeAndAfter {
+  private val originalTimeZone = TimeZone.getDefault
+  private val originalLocale = Locale.getDefault
+  private val testTempDir = Utils.createTempDir()
+
+  import org.apache.spark.sql.hive.test.TestHive.implicits._
+
+  override def beforeAll() {
+    TestHive.cacheTables = true
+    // Timezone is fixed to America/Los_Angeles for those timezone sensitive tests (timestamp_*)
+    TimeZone.setDefault(TimeZone.getTimeZone("America/Los_Angeles"))
+    // Add Locale setting
+    Locale.setDefault(Locale.US)
+
+    // The following settings are used for generating golden files with Hive.
+    // We have to use kryo to correctly let Hive serialize plans with window functions.
+    // This is used to generate golden files.
+    sql("set hive.plan.serialization.format=kryo")
```

Comment:
not related to this PR. 
maybe we should set kryo by default, otherwise there are many tests can not generate golden answer(i have tested it locally)